zerver/views/realm_aliases.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+from django.core.exceptions import ValidationError
+from django.db.utils import IntegrityError
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import has_request_variables, REQ, require_realm_admin
+from zerver.lib.actions import do_add_realm_alias, do_get_realm_aliases, \
+    do_remove_realm_alias
+from zerver.lib.response import json_error, json_success
+from zerver.lib.validator import check_bool, check_string
+from zerver.models import can_add_alias, RealmAlias, UserProfile
+
+from typing import Text
+
+def list_aliases(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> (HttpResponse)
+    aliases = do_get_realm_aliases(user_profile.realm)
+    return json_success({'aliases': aliases})
+
+@require_realm_admin
+@has_request_variables
+def create_alias(request, user_profile, domain=REQ(validator=check_string), subdomains_allowed=REQ(validator=check_bool)):
+    # type: (HttpRequest, UserProfile, Text, bool) -> (HttpResponse)
```

Comment:
@rishig I have been able to craft the following regular expression for checking valid domains: `^(?:[a-z0-9]+(?:[a-z0-9-]*[a-z0-9]{1})?\.)+(?:[a-z0-9]+(?:[a-z0-9-]*[a-z0-9]{1})?)$`? But after reading the code in `forms.RegistrationForm`, I think reusing that code will be a good idea since it tells what needs to be corrected as well as easy to understand.